marshaller.go

```diff
@@ -419,6 +438,40 @@ func marshalOrderByClause(v interface{}, tableName string, s interface{}) (strin
 
 }
 
+// v is the limit value provided
+func marshalLimitClause(v interface{}) (string, error) {
+	s, err := marshalIntValue(v)
+	if err != nil {
+		return "", ErrInvalidLimitClause
+	}
+	return s, nil
+}
+
+// v is the offset value provided
+func marshalOffsetClause(v interface{}) (string, error) {
+	s, err := marshalIntValue(v)
+	if err != nil {
+		return "", ErrInvalidOffsetClause
+	}
+	return s, nil
+}
+
+func marshalIntValue(v interface{}) (string, error) {
+	limit, ok := v.(int)
+	if !ok {
+		return "", errors.New("invalid type")
+	}
+	if limit < 0 {
+		return "", errors.New("invalid value")
+	}
+
+	var vString string
+	if limit > 0 {
```

Comment:
Since `marshalIntValue` will be used for both `offset` and `limit` this check would prevent us from having `OFFSET 0` in the query.

Reply:
I had assumed that the library wouldn't need to allow `LIMIT 0` or `OFFSET 0` and would rather leave them off.  I now see that those are both supported clauses in SOQL, so I guess I'll modify the type of the `limitClause` parameter to a `*int` so that the zero value is `nil` instead of `0`.  How would you feel about a `func Int(v int) *int` helper function for consumers to easily create pointers? I know the go AWS SDK provides similar functions, but I'm not sure if it would feel out of place here.

Reply:
You are right, let's only allow `int*` for `limitClause`. For `offsetClause` it is ok to have both `int` and `int*`.

Adding `func Int(v int) *int` will feel little out of place for this library. I think we will leave it to the users of library to create pointers.

Reply:
Also, please make sure you update README to include `limitClause` and `offsetClause` details. Adding it to `Advanced Usage` section is sufficient.